flutter_local_notifications/lib/src/platform_flutter_local_notifications.dart

```diff
@@ -349,42 +325,12 @@ class IOSFlutterLocalNotificationsPlugin
   /// except for when the time zone used in the [scheduledDate] matches the
   /// device's time zone and [uiLocalNotificationDateInterpretation] is set to
   /// [UILocalNotificationDateInterpretation.wallClockTime].
-  Future<void> zonedSchedule(
-    int id,
-    String title,
-    String body,
-    TZDateTime scheduledDate,
-    IOSNotificationDetails notificationDetails, {
-    @required
-        UILocalNotificationDateInterpretation
-            uiLocalNotificationDateInterpretation,
-    String payload,
-    ScheduledNotificationRepeatFrequency scheduledNotificationRepeatFrequency,
-  }) async {
-    validateId(id);
-    validateDateIsInTheFuture(scheduledDate);
-    ArgumentError.checkNotNull(uiLocalNotificationDateInterpretation,
-        'uiLocalNotificationDateInterpretation');
-    final Map<String, Object> serializedPlatformSpecifics =
-        notificationDetails?.toMap() ?? <String, Object>{};
-    await _channel.invokeMethod(
-        'zonedSchedule',
-        <String, Object>{
-          'id': id,
-          'title': title,
-          'body': body,
-          'platformSpecifics': serializedPlatformSpecifics,
-          'payload': payload ?? '',
-          'uiLocalNotificationDateInterpretation':
-              uiLocalNotificationDateInterpretation.index,
-        }
-          ..addAll(scheduledDate.toMap())
-          ..addAll(scheduledNotificationRepeatFrequency == null
-              ? <String, Object>{}
-              : <String, Object>{
-                  'scheduledNotificationRepeatFrequency':
-                      scheduledNotificationRepeatFrequency.index
-                }));
+  Future<void> zonedSchedule(List<NotificationData> notifications) async {
```

Comment:
Can you make bulk scheduling a method on it's own? Currently it's done as a breaking change when it's more of a helper/extension and this change would force everyone that schedules a single notification to pass a list

Reply:
isn't this class abstracted away from users? 

I've kept the signature as-is for `FlutterLocalNotificationsPlugin.zonedSchedule` for scheduling a single notification, and added [`FlutterLocalNotificationsPlugin.scheduleNotifications`](https://github.com/MaikuB/flutter_local_notifications/pull/683/files#diff-cda68032e4b223cfb99a263bd79ecf37R254) for multiple notifications.

Reply:
These classes aren't abstracted away by the cross-platform facing API if that's what you mean. Developers can choose to use the platform-specific implementation of the plugin to access methods that are specific to a platform. At a later point `zonedSchedule` will be part of the platform interface as well. Haven't done so yet as the beta release to get some feedback on this